Adding new HMM-based conservation pipeline.

davidjakubec opened the pull request:
Accelerated conservation calculations using HMMER. New pipeline files added to conservation_hmm/.

skodapetr commented:
Some general comments:
 * 80 characters per line
 * Use double quotes
 * Add README.md file into the directory with some description (installation step, or link to repository where the instalation steps are ... )

davidjakubec commented:
The changes are in! I have:
 - formatted `conservation_hmm.py` with black for ultra-orthodox PEP8 compliance;
 - added README.md with installation and other instructions;
 - renamed tests/ to examples/ and moved database/ there to improve clarity.

skodapetr commented on conservation_hmm/README.md:
```diff
@@ -0,0 +1,35 @@
+
+### Requirements
+
+The [HMMER](http://hmmer.org/) software package, including the Easel tools, must be installed and the programs must be available in `$PATH`. All development and testing is done using HMMER 3.3.2.
```
I would merge this to installation and add example on how to install on Debian.

skodapetr commented on conservation_hmm/README.md:
```diff
@@ -0,0 +1,35 @@
+
+The [HMMER](http://hmmer.org/) software package, including the Easel tools, must be installed and the programs must be available in `$PATH`. All development and testing is done using HMMER 3.3.2.
+
+The script `examples/database/download_Swiss-Prot.sh` can be used to download an example sequence database.
```
I would remove the script and rather put the command here directly. The script is small and I would not expect user to actually use it. Keep in mind that it now becomes part of a single package not a standalone application.

skodapetr commented on conservation_hmm/README.md:
````diff
@@ -0,0 +1,35 @@
+
+Use
+```
+conservation_hmm.py FASTA_file database_file working_directory target_file
````
All the arguments are positional, can't we use arguments like ```--fasta-file```, ```--working-directory```.
Also I would move this into the python script and here show example of commands to run, including the download command.

As user can then type ```conservation_hmm.py -h``` to get the help. That seems better to me, as I can get the help when and where I need it, in the terminal.

skodapetr commented on conservation_hmm/README.md:
```diff
@@ -0,0 +1,35 @@
+
+### Output
+
+When the script terminates, the `target_file` will contain a list of tab-separated triples \(index, IC, amino\_acid\_residue\) for the amino acid residues in the `FASTA_file`, where index is simply a number starting from zero \(0\) for the first residue. One triple is provided per line.
```
Again I would perhaps merge this into the use-case example with some example. 

The idea is that here in this document, the user can find a list of commands to run. So I can try it easily on my own.

skodapetr commented on conservation_hmm/conservation_hmm.py:
```diff
@@ -0,0 +1,99 @@
+
+def generate_MSA(FASTA_file, database_file, working_directory):
+    run(
+        "phmmer -o /dev/null -A {}{}.sto {} {}".format(
```
The Python we are using is 3.7, so there is already support for f-strings. I would highly recommend using them.
